packages/core/src/components/context-menu/contextMenu.tsx

```diff
@@ -229,6 +219,33 @@ export const ContextMenu: React.FC<ContextMenuProps> = React.forwardRef<any, Con
     );
 
     const containerClassName = classNames(className, Classes.CONTEXT_MENU);
+    const popoverElement = React.useRef<HTMLDivElement | null>(null);
+    const cancelContextMenu = React.useCallback((e: React.SyntheticEvent<HTMLDivElement>) => e.preventDefault(), []);
+
+    const popoverClasses = classNames(Classes.CONTEXT_MENU_POPOVER, popoverProps?.popoverClassName, {
+        [Classes.DARK]: isDarkTheme,
+    });
+    // compute an appropriate transform origin so the scale animation points towards target
+    const transformOrigin = getTransformOrigin(CONTEXT_MENU_PLACEMENT, undefined);
+
+    // only render the popover if there is content in the context menu;
+    // this avoid doing unnecessary rendering & computation
+    const maybePopover =
+        menuContent === undefined || targetOffset === undefined || !isOpen ? undefined : (
+            <PopoverOverlay
+                containerRef={popoverElement}
+                content={<div onContextMenu={cancelContextMenu}>{menuContent}</div>}
```

Comment:
previously, this `cancelContextMenu` functionality happened in `ContextMenuPopover`